src-tests/test-rpc.lisp

```diff
@@ -101,6 +101,30 @@
         ;; kill the server thread
         (kill-thread-slowly server-thread)))))
 
+(deftest test-RPCRequest-client_timeout-overrides-server-timeout ()
+  (with-unique-rpc-address (addr)
+    (let* ((server-function
+             (lambda ()
+               (let ((dt (rpcq:make-dispatch-table)))
+                 (rpcq:dispatch-table-add-handler dt (lambda () (sleep 2)) :name "test_method")
+                 (rpcq:start-server :dispatch-table dt
+                                    :listen-addresses (list addr)))))
+           (server-thread (bt:make-thread server-function)))
+      (sleep 1)
+      (unwind-protect
+           (let* ((id (format nil "~A" (uuid:make-v4-uuid)))
+                  (request (make-instance 'rpcq::|RPCRequest|
+                                          :|method| "test_method" :|params| (make-hash-table)
+                                          :|id| id :|client_timeout| 1)))
+             (rpcq:with-rpc-client (client addr)
+               (signals rpc-error
+                 (rpcq::%rpc-call-raw-request
```

Comment:
We should just update `rpc-call` to also pass along the client timeout, no?

Reply:
I don't think so. That would mean changing the signature of `rpc-call`. Constructing the `RPCRequest` manually and passing it along this way seems like an appropriate use of `rpc-call`. An alternative would be something similar to `rpc-call` but which takes a `RPCRequest`.

Reply:
The client timeout is saved in a slot of the `client` struct. I don't think the `rpc-call` signature would have to change?

Reply:
I don't follow.

Reply:
something like this (untested)

```lisp
(defun rpc-call (client call &rest args)
  "Makes a synchronous RPC call, designated by the string method name CALL, over the connection CLIENT.  ARGS is a plist of arguments.  Returns the result of the call directly."
  (let* ((uuid (format nil "~a" (uuid:make-v4-uuid)))
         (request (make-instance '|RPCRequest|
                                 :|id| uuid
                                 :|params| (prepare-rpc-call-args args)
                                 ;; only line of diff is here
                                 :|client_timeout| (rpc-client-timeout client)
                                 :|method| (sanitize-name call)))
         (payload (serialize request)))
    (%rpc-call-raw-request client uuid payload)))
```

Reply:
Then the test could just use `with-rpc-client` + `rpc-call` like so

```lisp
(rpcq:with-rpc-client (client addr :timeout 1)
  (signals rpcq::rpc-error
    (rpcq:rpc-call client "test-method" :sleep 5)))
```

(and using the pre-defined `test-method` that accepts a `:sleep` arg in this case)

Reply:
Gotya. Good idea.